### PR TITLE
Sprite sort refactor

### PR DESCRIPTION
**Pull request by janisozaur**
This is purely refactor without any side effects other than exposing a function so it may be used in tests.

### CODE REVIEW

File: src/openrct2/paint/paint.c [hunk 1]
```diff
@@ -733,7 +733,35 @@ void paint_generate_structs(rct_drawpixelinfo * dpi)
     }
 }
 
-static paint_struct * paint_arrange_structs_helper(paint_struct * ps_next, uint16 quadrantIndex, uint8 flag)
+static bool compareSpriteBBoxes(uint8 rotation, const paint_struct_bound_box *initialBBox,
```
Comment (ZehMatt) [inline src/openrct2/paint/paint.c]: I would probably call this something like isBBoxIntersecting as its not related to any sprite.

File: src/openrct2/paint/paint.h [hunk 2]
```diff
@@ -123,6 +123,15 @@ typedef struct sprite_bb {
     rct_xyz16 bb_size;
 } sprite_bb;
 
+typedef struct paint_struct_bound_box {
```
Comment (ZehMatt) [inline src/openrct2/paint/paint.h]: Maybe its better to move the struct into paint.h and also replace the members of paint_struct with it, later we can make a simple copy from it.

Comment (ZehMatt) [inline src/openrct2/paint/paint.h]: You could also replace the members of paint_struct with the new struct.

Comment (janisozaur) [reply src/openrct2/paint/paint.h]: Yes, this is coming in future commit.

**Pull request merged**